From 1e2081ee68d6d5629a55adab34c2b2ee4bd54cf6 Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Tue, 18 Oct 2016 11:09:33 +0200
Subject: [PATCH 6/8] mali: fix mali_gp_job_create page domain fault

For complex draw (like glmark2) using position buffer or varying buffer
bigger than 4096 Bytes, deferred bind memory mechanism is put in place.
In such cases page fault domain were observed due to a bad usage of
variable comming from userland.
Do not directly use "uargs" but "jog->uargs" issued by the copy from
user.

Change-Id: Ic471d70075529dfb9b568d084a89a93e62396f1a
Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
Reviewed-on: https://gerrit.st.com/59230
---
 src/devicedrv/mali/common/mali_gp_job.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/devicedrv/mali/common/mali_gp_job.c b/src/devicedrv/mali/common/mali_gp_job.c
index 8dd19cb..fb8dcd8 100755
--- a/src/devicedrv/mali/common/mali_gp_job.c
+++ b/src/devicedrv/mali/common/mali_gp_job.c
@@ -133,7 +133,7 @@ struct mali_gp_job *mali_gp_job_create(struct mali_session_data *session, _mali_
 				goto fail1;
 			}
 
-			memory_list = (u32 __user *)(uintptr_t)uargs->deferred_mem_list;
+			memory_list = (u32 __user *)(uintptr_t)job->uargs.deferred_mem_list;
 
 			if (0 != _mali_osk_copy_from_user(job->varying_list, memory_list, sizeof(u32) * job->uargs.deferred_mem_num)) {
 				MALI_PRINT_ERROR(("Mali GP job: Failed to copy varying list from user space!\n"));
@@ -173,7 +173,7 @@ struct mali_gp_job *mali_gp_job_create(struct mali_session_data *session, _mali_
 				}
 			}
 
-			if (uargs->varying_memsize > MALI_UK_BIG_VARYING_SIZE) {
+			if (job->uargs.varying_memsize > MALI_UK_BIG_VARYING_SIZE) {
 				job->big_job = 1;
 			}
 		}
-- 
2.7.4

